src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -734,7 +734,21 @@ private IncrementalHash PrepareDigest(bool compatMode)
 
         private HashAlgorithmName GetDigestAlgorithm()
         {
-            return PkcsHelpers.GetDigestAlgorithm(DigestAlgorithm.Value);
+            HashAlgorithmName digestAlgorithm = PkcsHelpers.GetDigestAlgorithm(
+                DigestAlgorithm.Value,
+                out string[] expectedSignatureAlgorithmOids,
+                forVerification: true);
+            if (expectedSignatureAlgorithmOids != null &&
```

Comment:
This could probably use a comment.  Something like

// Allow a signature algorithm identifier to be used as a digest algorithm identifier,
// but only when it matches the actual signature algorithm